homeassistant/components/lg_thinq/entity.py

```diff
@@ -106,3 +110,36 @@ async def async_call_api(
             if on_fail_method:
                 on_fail_method()
             raise ServiceValidationError(exc) from exc
+
+    def check_control_condition(
+        self, switch: str | None = None, state_condition: list[str] | None = None
+    ) -> bool:
+        """For determining control conditions according to device's state."""
+        if self.device_state is None:
+            # There are no restrictions on controlling devices that do not have a current_state
+            return True
+        # Check 1)is_on, 2)remote_control_enabled, 3)state_condition
+        if switch == SERVICE_TURN_OFF:
+            return True
+        if switch == SERVICE_TURN_ON and not self.device_state.power_on_enabled:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="power_on_command_not_supported",
+            )
+        if switch != SERVICE_TURN_ON and not self.device_state.device_is_on:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="command_not_supported_in_power_off",
+            )
```

Comment:
And I am wondering, this is entity.py and now we have logic for the switch entities to decide if it should work or not in a generic class. Should we overwrite(/extend) this in switch.py?